### PR TITLE
Ability to turn off Optifine warning and wiki URL in mcmod.info

### PR DESCRIPTION
**Pull request by josh-65**

### CODE REVIEW

File: src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java [hunk 1]
```diff
@@ -11,32 +11,31 @@
 import net.minecraftforge.fml.common.Mod;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.minecraftforge.fml.relauncher.Side;
+import net.tardis.mod.config.TardisConfig;
 
 import java.io.IOException;
 import java.util.List;
 
+import javax.lang.model.util.ElementScanner6;
```
Comment (Swirtzly) [inline src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java]: Why is this import needed?

File: src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java [hunk 2]
```diff
@@ -11,32 +11,31 @@
...
 
 	public GuiOptifineWarning() {
 	}
-	
+
 	@Override
 	public void initGui() {
-		super.initGui();
-		
-		this.addButton(new GuiButtonExt(0, this.width / 2 - 154, this.height - 48, 150, 20, I18n.format("Close Game")));
```
Comment (Swirtzly) [inline src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java]: Why has the close game button been removed?

File: src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java [hunk 3]
```diff
@@ -61,12 +60,17 @@ public void drawScreen(int mouseX, int mouseY, float partialTicks) {
 		@SubscribeEvent
 		public static void onLoadMC(GuiScreenEvent.InitGuiEvent e) {
 			if (!opened && FMLClientHandler.instance().hasOptifine() && e.getGui() instanceof GuiMainMenu) {
-				lastGui = e.getGui();
-				Minecraft.getMinecraft().displayGuiScreen(new GuiOptifineWarning());
-				opened = true;
+				{
+					if (TardisConfig.MISC.warn==true) {
```
Comment (Swirtzly) [inline src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java]: Why `if (TardisConfig.MISC.warn==true)` and not just `if (TardisConfig.MISC.warn)`

File: src/main/java/net/tardis/mod/config/TardisConfig.java [hunk 4]
```diff
@@ -97,6 +97,10 @@
 		@Config.LangKey("config.misc.recharge")
 		public float artronRechargeRate = 4F;
 
+		@Config.LangKey("config.misc.warn")
+		@Config.Comment("If this is enabled and optifine is detected, then a warning at startup is displayed")
+		public boolean warn = true;
```
Comment (Swirtzly) [inline src/main/java/net/tardis/mod/config/TardisConfig.java]: Should be named better, as seen in my reply to the PR

**Comment by Swirtzly** [comment]
The entire reason this functionality exists without a config in-game, is to stop people reporting the issues they have with optifine to us, so they get told each time so that we don't have to repeat ourselves every time someone has a rendering issue while having optifine installed.

The PR itself, is for @Spectre0987 to decide to merge

It should be noted that we value your contribution, but it is missing a lang file entry, "warn" is not very descriptive in code and it should be something more along the lines of "optifineWarning"

File: src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java (hunk 1 above)
Comment (josh-65) [reply src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java]: Didn't add it. I think visual studio code did when it tried auto placing code.

File: src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java (hunk 2 above)
Comment (josh-65) [reply src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java]: I hit it sometimes when clicking on the window and there's no point of it as people can just hit the normal games quit.

File: src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java (hunk 3 above)
Comment (josh-65) [reply src/main/java/net/tardis/mod/client/guis/GuiOptifineWarning.java]: Didn't know i could i'm new to trying java.

**Comment by josh-65** [comment]
Thanks for the feedback. Forgot to upload the lang file i edited.
